### DISCUSSION TITLE
scanning of test to practice

### DISCUSSION THREAD

**Discussion by solidity-cobra** [Cyfrin/foundry-full-course-cu#1461] [General]
please how do i scan the code to be able to take the test at the end of each lessons

**Comment by EngrPips** [comment]
hello @solidity-cobra , You should use a QR code scanner to scan the code and actually most lessons have links that would take you straight to the test instead of having to scan some QR code, maybe you should look for those instead.

**Reply by solidity-cobra** [reply]
i don,t even undertstand the test, how did you go about it

**Reply by EngrPips** [reply]
Hello @solidity-cobra  You are to call the contract with the appropriate argument and if you passed in the appropriate argument you would have solved the challenge and you will be minted an NFT that serves as a badge for finishing that section of the course. If you still find anything confusing please don't hesitate to ask.

**Reply by solidity-cobra** [reply]
it shows enter a number and my twitter handle, is that the challenge?

**Comment by solidity-cobra** [comment]
![Screenshot (11)](https://github.com/Cyfrin/foundry-full-course-f23/assets/161164747/f401b2a9-44a9-4b07-8f3e-986bb0c843bb)
check this @EngrPips

**Reply by EngrPips** [reply]
@solidity-cobra  seem like you solved the challenge, congratulations.

**Reply by solidity-cobra** [reply]
thanks @EngrPips 
at the end of the course, are we awarded certificates?
if no, how can we get one

**Reply by EngrPips** [reply]
Hello @solidity-cobra, as of this moment there isn't a certificate for finishing the course but i heard there is something about issuing certificate in the pipeline.

**Reply by solidity-cobra** [reply]
@EngrPips  my remix has stopped compiling

**Comment by EngrPips** [comment]
Hello @solidity-cobra , hmmmm did you get any error from remix.

**Reply by solidity-cobra** [reply]
no error, it doesn't do anything or even show me warnings. once i click compile, it takes forever and never compile. i noticed this today as i decided to practice.
it was working fine.

**Reply by EngrPips** [reply]
Are you sure your internet connection is strong ?, network interruption might cause such issues

**Reply by solidity-cobra** [reply]
ohhh, it might be.
i will check it out thanks.

**Reply by solidity-cobra** [reply]
are you a student like us or you are a full time developer?

**Reply by EngrPips** [reply]
Yeah please check and let's know if that is the cause, I am a full time developer. I graduated from university two years ago.

**Reply by solidity-cobra** [reply]
its compiling now, it was actually caused by the poor network connection.
thank you.
to be honest, coding isn't easy, i always feel like i don't know what i'm doing

**Reply by EngrPips** [reply]
I am glad it is working now and yeah nothing is easy when you first start doing it. it's how many years now you started coding ?

**Reply by solidity-cobra** [reply]
I just started, wanted learning python but i didn't have a road map. so when i came across this course, i decided to give it a try.

**Reply by EngrPips** [reply]
You see that is the thing with we human, we complain a lot and do very little then expect a mighty reward (including myself). i mean you just started and here you are complaining it is hard. i don't mean to be harsh or say you shouldn't be complaining but then i want you know that becoming anything meaningful, or gaining any knowledge that would be of great benefit is only through hardship, suffering and difficulty.  So choose a path that you are truly interested in and want to pursue then make up your mind that no matter what you would see it to the absolute end. 

And you know what is even more good there is whole community to always help get through your difficult times, so be happy and keep going

**Reply by solidity-cobra** [reply]
hello @EngrPips  i'm having issues, i cant get the interface use in lesson 4. i followed the video and when i got to the smartcontractkit in github, i clicked contract/src,interface,v8.0 and i dint see aggregatorv3.sol

**Comment by EngrPips** [comment]
Hello @solidity-cobra , Lesson 4 is `REMIX FUND ME` so you should be importing the `AggregatorV3Interface` straight from chainlink github repo to the remix environment. maybe you are not doing something correctly. below is a picture that shows me successfully importing an `AggregatorV3Interface`

<img width="1059" alt="Screenshot 2024-03-13 at 1 12 06 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/42423680-7605-435e-aa4a-2cd3770213f0">

**Reply by solidity-cobra** [reply]
please help me with the link to aggretorV3Interface.
i cant get access to it

**Reply by solidity-cobra** [reply]
@EngrPips  thanks, i have figured it out

**Reply by EngrPips** [reply]
You are always welcome

**Comment by solidity-cobra** [comment]
![Screenshot (12)](https://github.com/Cyfrin/foundry-full-course-f23/assets/161164747/668c236a-1e74-4e33-b697-f989be458665)
@EngrPips  please check this out

**Comment by EngrPips** [comment]
Hello @solidity-cobra , here is the link `https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol` 

**A PICTURE OF THE CONTRACT ON CHAINLINK REPO**

<img width="1433" alt="Screenshot 2024-03-13 at 3 55 53 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/422cef93-41e6-49ca-9ce2-ddca179871b9">

**Reply by solidity-cobra** [reply]
thanks @EngrPips

**Reply by EngrPips** [reply]
Anytime friend

**Comment by solidity-cobra** [comment]
@EngrPips hello sir, hope you are doing great today?
i'm in need of your help sir.
please i need clarification on whether to know this codes of hand or to understand it as a beginner, i am asking because sometimes i forget this codes but when i see them, i know what they mean. 
sir please as a senior dev,  i will like to know how you passed through this stage.

**Reply by EngrPips** [reply]
Hello @solidity-cobra , you are thinking more of me than i am lol but honestly you don't need to know any code off hand. i mean it's fine that you forget stuffs and you don't remember everything and you need to check stuffs or do some browsing. That's totally normal as far as you know what the code does knowing the syntax off hand is not just important at all.

**Reply by solidity-cobra** [reply]
thank you sir

**Comment by solidity-cobra** [comment]
hello @EngrPips , hope you are doing great today?
please i'm having issues with my vscode.
it keeps loading opening remote

**Reply by EngrPips** [reply]
Hello @solidity-cobra , I am doing great and it's great to hear from you friend. Have you checked if your vscode needs update ?

**Reply by solidity-cobra** [reply]
how do i check that?
i just installed it recently

**Reply by EngrPips** [reply]
It is suppose to inform you if there is an update that needs to be done actually. i would suggest you browse google for what could be causing that issue and the possible fixes. maybe you would find something that would help

**Reply by Alhassan78** [reply]
![Screenshot (1)](https://github.com/Cyfrin/foundry-full-course-f23/assets/89581088/dfddf2cd-27be-4cbb-8527-0395a2164370)
Please @EngrPips , I'm trying to run "forge init"
And it gives error saying "cannot run 'ini' on a non-empty directory"
My directory is empty

**Reply by EngrPips** [reply]
Hello @Alhassan78 , I am not really familiar with `wsl` as i have never used it before but then i would suggest you open a new and empty directory and try `forge init` inside of it, i believe that should work

**Reply by Alhassan78** [reply]
Alright

**Comment by solidity-cobra** [comment]
hello @EngrPips  i'm back.
i just want to know if you are still here sir.
its been a while

**Reply by EngrPips** [reply]
Indeed, I am still here. Welcome back, Friend.

**Reply by solidity-cobra** [reply]
thank you

**Reply by EngrPips** [reply]
You are welcome.
